### PR TITLE
Edited doc to remove fluff and snark

### PR DESCRIPTION
**Pull request by IgnatiusFlo**
I edited the doc to conform to the 80 character line rule. Also, copy editing on the text.
@Ian-Foote

### CODE REVIEW

File: docs/ref/settings.txt [hunk 1]
```diff
@@ -37,7 +37,8 @@ a model object and return its URL. This is a way of inserting or overriding
         'news.story': lambda o: "/stories/%s/%s/" % (o.pub_year, o.slug),
     }
 
-Note that the model name used in this setting should be all lower-case, regardless
+Note that the model name used in this setting should be all lower-case, regard-
+less
 of the case of the actual model class name.
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: I think I'd write this as:

```
Note that the model name used in this setting should be all lower-case,
regardless of the case of the actual model class name.
```

File: docs/ref/settings.txt [hunk 2]
```diff
@@ -82,7 +85,8 @@ Django also allows the `fully qualified domain name (FQDN)`_ of any entries.
 Some browsers include a trailing dot in the ``Host`` header which Django
 strips when performing host validation.
 
-.. _`fully qualified domain name (FQDN)`: https://en.wikipedia.org/wiki/Fully_qualified_domain_name
+.. _`fully qualified domain name (FQDN)`: https://en.wikipedia.org/wiki/
+Fully_qualified_domain_name
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: I think breaking a link in two is worse than having a long line.

File: docs/ref/settings.txt [hunk 3]
```diff
@@ -2120,7 +2128,7 @@ affect them.
 Default: ``False``
 
 If ``True``, the :class:`~django.middleware.security.SecurityMiddleware` sets
-the :ref:`x-xss-protection` header on all responses that do not already have it.
+the :ref:`x-xss-protection` header on all responses that do not already have it
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: Why did you remove the `.`?

File: docs/ref/settings.txt [hunk 4]
```diff
@@ -2222,16 +2230,14 @@ available in ``request.META``.)
 
 .. warning::
 
-    **You will probably open security holes in your site if you set this
-    without knowing what you're doing. And if you fail to set it when you
-    should. Seriously.**
+    **These settings can compromise site security**
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: I think you've changed the meaning here. Both setting and not setting these can cause a security problem.

File: docs/ref/settings.txt [hunk 5]
```diff
@@ -2642,7 +2649,7 @@ See also :setting:`DECIMAL_SEPARATOR`, :setting:`NUMBER_GROUPING` and
 Default: ``False``
 
 A boolean that specifies if datetimes will be timezone-aware by default or not.
-If this is set to ``True``, Django will use timezone-aware datetimes internally.
+If this is set to ``True``, Django will use timezone-aware datetimes internally
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: You've lost the `.` here.

File: docs/ref/settings.txt [hunk 6]
```diff
@@ -2897,13 +2904,14 @@ Controls where Django stores message data. Valid values are:
 * ``'django.contrib.messages.storage.session.SessionStorage'``
 * ``'django.contrib.messages.storage.cookie.CookieStorage'``
 
-See :ref:`message storage backends <message-storage-backends>` for more details.
+See :ref:`message storage backends <message-storage-backends>` for more details
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: You've lost the `.` here.

**Comment by Ian-Foote** [comment]
@adamchainz

File: docs/ref/settings.txt [hunk 7]
```diff
@@ -3080,12 +3086,9 @@ Whether to use a secure cookie for the session cookie. If this is set to
...
 
-Since it's trivial for a packet sniffer (e.g. `Firesheep`_) to hijack a user's
-session if the session cookie is sent unencrypted, there's really no good
-excuse to leave this off. It will prevent you from using sessions on insecure
-requests and that's a good thing.
-
-.. _Firesheep: http://codebutler.com/firesheep
+It's trivial for a packet sniffer to hijack a user's
+session if the session cookie is sent unencrypted. It will prevent you from
+using sessions on insecure requests.
```
Comment (adamchainz) [inline docs/ref/settings.txt]: The "it" in the second sentence is now unclear How about changing to "Turning this setting on will prevent this as cookie will never be sent unencrypted."

**Comment by IgnatiusFlo** [comment]
Hi Tim,
Sure. I was doing this as a beginner contribution in the sprint at Pycon. Ian kindly walked me through the changes and we reverted all the cosmetic ones. 
Cheers.

File: docs/ref/settings.txt [hunk 8]
```diff
@@ -3001,9 +2995,7 @@ useful way to mitigate the risk of a client side script accessing the
 protected cookie data.
 
 Turning it on makes it less trivial for an attacker to escalate a cross-site
-scripting vulnerability into full hijacking of a user's session. There's not
```
Comment (timgraham) [inline docs/ref/settings.txt]: I don't see much advantage to removing the sentence here and in SESSION_COOKIE_SECURE. It eliminates the answer to the question, "is there a use case for disabling?" If the answer is "no", then perhaps there's an argument for removing the setting, but that's another matter.

Any objections? If not, I have a few other edits that I'll push along with this (mainly, some rewrapping closer to 79 characters for the modified lines).

Comment (Ian-Foote) [reply docs/ref/settings.txt]: I think if we're going to recommend never turning this off we should be more direct than we were before. Without knowledge of any use-cases I'm in favour of deprecation and removal.

File: docs/ref/settings.txt [hunk 9]
```diff
@@ -3001,9 +2995,7 @@ useful way to mitigate the risk of a client side script accessing the
...
-much excuse for leaving this off, either: if your code depends on reading
-session cookies from JavaScript, you're probably doing it wrong.
+scripting vulnerability into full hijacking of a user's session.
 
 .. _HTTPOnly: https://www.owasp.org/index.php/HTTPOnly
```
Comment (Ian-Foote) [inline docs/ref/settings.txt]: This link looks very out of date, containing only [minor changes since 2014](https://www.owasp.org/index.php?title=HttpOnly&type=revision&diff=232658&oldid=185263). Perhaps https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Secure_and_HttpOnly_cookies would be more appropriate if we keep the setting?

It feels like this is getting out of scope of this PR though, so perhaps creating a ticket would be best.

Comment (adamchainz) [reply docs/ref/settings.txt]: There are a bunch of old links like this, some needing minor updates e.g. to be HTTPS over HTTP, others like this probably better to re-point. I think it would be a separate ticket?

Comment (IgnatiusFlo) [reply docs/ref/settings.txt]: I could try and do that. If it's better to have a seperate ticket would you assign it to me?

Comment (Ian-Foote) [reply docs/ref/settings.txt]: I've created https://code.djangoproject.com/ticket/29784.

Comment (carltongibson) [reply docs/ref/settings.txt]: @IgnatiusFlo Ticket is accepted. If you want to grab it that would be fab! (c.f. #10430)

Comment (IgnatiusFlo) [reply docs/ref/settings.txt]: Hey, Sure. Sorry. It looked like someone had taken it. I won't be in front of my computer till tomorrow (no allowed python on my work PC), so bear with me....

Comment (IgnatiusFlo) [reply docs/ref/settings.txt]: Or can I just edit it in my fork? Do I need anything else?

File: docs/ref/settings.txt [hunk 10]
```diff
@@ -37,7 +37,7 @@ a model object and return its URL. This is a way of inserting or overriding
         'news.story': lambda o: "/stories/%s/%s/" % (o.pub_year, o.slug),
     }
 
-Note that the model name used in this setting should be all lower-case, regardless
+Note that the model name used in this setting should be all lowercase, regardless
```
Comment (jdufresne) [inline docs/ref/settings.txt]: The spelling "lower-case" appears several times throughout the project. Should we normalize this spelling globally?

```
./docs/ref/models/fields.txt:1682:the ``OneToOneField``, Django will use the lower-case name of the current model
./docs/ref/settings.txt:40:Note that the model name used in this setting should be all lower-case, regardless
./docs/topics/db/models.txt:984:- ``'%(class)s'`` is replaced by the lower-cased name of the child class
./docs/topics/db/models.txt:986:- ``'%(app_label)s'`` is replaced by the lower-cased name of the app the child
./docs/topics/db/models.txt:1068:``Place`` object to the ``Restaurant`` object by using the lower-case version
./docs/topics/i18n/index.txt:77:      are generally represented in lower-case, but the HTTP ``Accept-Language``
./tests/many_to_one/models.py:47:# created (the field names being lower-cased versions of their opposite
./django/http/request.py:541:    Returned domain is lower-cased. If the host is invalid, the domain will be
./django/http/request.py:569:    Note: This function assumes that the given host is lower-cased and has
./django/http/response.py:39:        # _headers is a mapping of the lower-case name to the original case of
./django/contrib/gis/feeds.py:62:                # Getting the lower-case geometry type.
./django/contrib/gis/gdal/geomtype.py:29:    # Reverse type dictionary, keyed by lower-case of the name.
./django/db/models/fields/reverse_related.py:152:        # many-to-many objects. It uses the lower-cased object_name + "_set",
```

Let me know if you'd like to do this, otherwise I don't mind.

Comment (adamchainz) [reply docs/ref/settings.txt]: I say we normal-ize it, it's worth doing in a separate pull-request

Comment (jdufresne) [reply docs/ref/settings.txt]: Done in  #10440.

**Comment by jdufresne** [comment]
@IgnatiusFlo Do you mind rebasing this to resolve the merge conflicts with recent commits?

**Comment by timgraham** [comment]
I'll do it when I push my edits.

**Comment by IgnatiusFlo** [comment]
Hi, I'm really sorry but I have such limited time to to any of this stuff it's probably best I just bow out. Sorry if I've wasted anyone's time.

**Comment by Ian-Foote** [comment]
@IgnatiusFlo That's absolutely ok and you haven't wasted any of my time.

**Comment by timgraham** [comment]
I'll push the edits to this PR.

**Comment by adamchainz** [comment]
👍
